java/com/google/flatbuffers/Utf8.java

```diff
@@ -73,6 +77,56 @@ public static void setDefault(Utf8 instance) {
     DEFAULT = instance;
   }
 
+  /**
+   * Encode a Java's CharSequence UTF8 codepoint into a byte array.
+   * @param in CharSequence to be encoded
```

Comment:
this looks like a copy of the code in `Utf8Safe.encodeUtf8Buffer`.. can this code be shared with no performance consequences?

Reply:
@aardappel I guess it would be possible to modify `encodeUtf8CodePoint(CharSequence in, int start, byte[] out)` to `encodeUtf8CodePoint(CharSequence in, ByteBuffer out)`

And make `Utf8Safe.encodeUtf8Buffer` call `encodeUtf8CodePoint` for each codepoint in `CharSequence in`. But I guess there will be small performance degradation in both methods. Not sure if enough to justify having it separate or not. We could:

1. Keep it separate as in the MR, "optimal"  in performance on both methods 
2. Try to reuse `encodeUtf8CodePoint` inside `encodeUtf8Buffer` might slightly lose performance
3. Keep as it for now, but reuse later if we see the need for the same pattern in other areas of the code.

Up to you.

Reply:
If you think it has performance consequences, then lets leave it as-is for the moment. Can be cleaned up later if someone can show the difference is marginal.